server/resources/CampaignResources.py

```diff
@@ -39,17 +27,20 @@ def get(self, id):
         prospects = []
         for prospect in current_campaign.prospects:
             prospects.append({'id' : prospect.id, 'name' : prospect.name})
+        steps = []
+        for step in current_campaign.steps:
+            steps.append({'id' : step.id, 'email_template_id' : step.email_template.id})
         return {
-            'Campaign': current_campaign.name,
-            'Prospects': prospects
+            'Campaign' : current_campaign.name,
+            'Prospects' : prospects,
+            'Steps' : steps
             }, 200 
-        
-class AddProspectsToCampaign(Resource):
+
     @jwt_required
-    @validate_args("prospect_ids", "campaign_id")
```

Comment:
The validate_args decorator is cool, but but I think we might be better off using the built-in validation in reqparse. For the modules that have multiple resources, you can have multiple parsers, which would make it more clear what the required and optional parameters are for each resource. With all of the arguments in one parser and a validator that takes any number of those, it would be hard to tell if there are optional args without looking through the whole method to see what args are used. add_argument has the optional required parameter, as well as ways to add error messages.
[https://flask-restful.readthedocs.io/en/0.3.5/reqparse.html#error-handling](https://flask-restful.readthedocs.io/en/0.3.5/reqparse.html#error-handling)

Reply:
I tried looking into the built-in validation in reqpars, but it seems that if you set a global parser in one file and make one of the arguments required, all the post methods in that file are required to have that argument as an input. Unless we can make multiple parser somehow and have them separate, what's your method of doing so?

Reply:
Yeah, you can just make more than one RequestParser. Like you could have new_campaign_parser and add_prospects_parser that each have their own set of arguments, and just call parse_args() on the one you need in that method.

Reply:
Ah right, I see. But idk if Shums likes that the fact we have to spam code, idk we can bring it up during today's standup.